snooty/types.py

```diff
@@ -233,19 +233,21 @@ def open(cls, root: Path) -> Tuple["ProjectConfig", List[Diagnostic]]:
         diagnostics: List[Diagnostic] = []
         while path.parent != path:
             try:
-                with path.joinpath("snooty.toml").open("rb") as f:
-                    data = tomli.load(f)
-                    data["root"] = path
-                    result, parsed_diagnostics = check_type(
-                        ProjectConfig, data
-                    ).render_constants()
+                toml_text = path.joinpath("snooty.toml").read_text()
+                data = util.parse_toml_and_add_line_info(toml_text)
+                data["root"] = path
+                result, parsed_diagnostics = check_type(
+                    ProjectConfig, data
+                ).render_constants()
 
-                    parsed_diagnostics.extend(cls.validate_data(result.data))
+                parsed_diagnostics.extend(cls.validate_data(result.data))
 
-                    return result, parsed_diagnostics
+                return result, parsed_diagnostics
             except FileNotFoundError:
                 pass
-            except LoadError as err:
+            except util.TOMLDecodeErrorWithSourceInfo as err:
+                diagnostics.append(UnmarshallingError(str(err), err.lineno))
+            except (LoadError, tomli.TOMLDecodeError) as err:
```

Comment:
```suggestion
            except LoadError as err:
```

since we already caught `TOMLDecodeError` within `parse_toml_and_add_line_info`